### PR TITLE
OIDC: Redirect to delegated auth provider when signing out

### PR DESCRIPTION
**Pull request by kerryarchibald**
Fixes https://github.com/vector-im/element-web/issues/26000

In the case of OIDC-aware clients, you cannot sign out of _other_ sessions via the device manager in EW anymore. Instead, users will need to be guided towards the MAS UI to take this actions.

[MSC3824](https://github.com/matrix-org/matrix-spec-proposals/pull/3824) now describes this requirement.

**ACs**

- When the HS has a MAS deployment configured:
  - **Clicking the sign-out button in the EW device manager for sessions _other_ than the current redirects to MAS**
  - **There is an interstitial dialog that explains that the user will be redirected to perform the action on MAS**
  - **The redirection lands the user on a deep-link (query hash as per the MSC) into the respective session's details page on MAS**
    - Making the deep link URL actually work is not part of this issue
  - **Multi-session signout is disabled**

Open confirm dialog, continue is a link to the auth provider opening in a new tab
Screenshot 2023-08-21 at 20 43 19

After clicking continue, wait for user to return from the other tab, then refresh session list when they close the modal
Screenshot 2023-08-21 at 20 43 31

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * OIDC: Redirect to delegated auth provider when signing out ([\#11432](https://github.com/matrix-org/matrix-react-sdk/pull/11432)). Fixes vector-im/element-web#26000. Contributed by @kerryarchibald.

### CODE REVIEW

File: src/components/views/dialogs/QuestionDialog.tsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ export interface IQuestionDialogProps {
     onFinished(ok?: boolean): void;
 }
 
-export default class QuestionDialog extends React.Component<IQuestionDialogProps> {
+export default class ConfirmRedirectDialog extends React.Component<IQuestionDialogProps> {
```
Comment (t3chguy) [inline src/components/views/dialogs/QuestionDialog.tsx:42]: This doesn't seem right, should match the filename

File: src/components/views/dialogs/oidc/OidcLogoutDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2015, 2016 OpenMarket Ltd
+Copyright 2017 New Vector Ltd.
```
Comment (t3chguy) [inline src/components/views/dialogs/oidc/OidcLogoutDialog.tsx:3]: This seems wrong

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: src/components/views/dialogs/QuestionDialog.tsx (hunk 1 above)
Comment (kerryarchibald) [reply src/components/views/dialogs/QuestionDialog.tsx]: Yikes, sloppy!
Fixed

**Comment by kerryarchibald** [comment]
@t3chguy I've enabled auto merge, feel free to disable if you think it needs another look.

**Review by t3chguy — APPROVED** [review]

**Comment by hughns** [comment]
Disabling auto-merge as the param names have changed.

**Comment by t3chguy** [comment]
@andybalaam this is a release blocker that needs work

**Comment by hughns** [comment]
I've pushed changes to the param names. Ready for review again.

**Review by t3chguy — APPROVED** [review]

**Comment by andybalaam** [comment]
@hughns looks like a unit test failed somewhere?

**Comment by andybalaam** [comment]
@t3chguy requesting a quick re-review of the new commits

**Review by t3chguy — APPROVED** [review]

**Pull request merged**